services/todo-legacy/controllers/todo-controller.api.ts

```diff
@@ -0,0 +1,180 @@
+              },
+              title: {
+                type: "string",
+                description: "The title of the Todo entry."
+              },
+              body: {
+                type: "string",
+                description: "The body of the Todo entry."
+              }
+            }
+          }
+        }
+      }
+    },
+    "/todo/{id}": {
+      put: {
+        "x-operation-name": "replaceTodo",
+        parameters: [
+          {
+            name: "id",
+            in: "path",
+            description: "The todo ID.",
+            required: true,
+            type: "string",
+            format: "JSON"
+          },
+          {
+            name: "data",
+            in: "body",
+            description: "The Todo model instance.",
+            required: true,
+            format: "JSON"
+          }
+        ],
+        responses: {
+          202: {
+            schema: {
+              id: {
+                type: "number",
+                description: "The ID number of the Todo entry."
+              },
+              title: {
+                type: "string",
+                description: "The title of the Todo entry."
+              },
+              body: {
+                type: "string",
+                description: "The body of the Todo entry."
+              }
+            }
+          }
+        }
+      },
+      patch: {
+        "x-operation-name": "updateTodo",
+        parameters: [
+          {
+            name: "id",
+            in: "path",
+            description: "The todo ID.",
+            required: true,
+            type: "string",
+            format: "JSON"
+          },
+          {
+            name: "data",
+            in: "body",
+            description: "The Todo model instance.",
+            required: true,
+            format: "JSON"
+          }
+        ],
+        responses: {
+          202: {
+            schema: {
+              id: {
+                type: "number",
+                description: "The ID number of the Todo entry."
+              },
+              title: {
+                type: "string",
+                description: "The title of the Todo entry."
+              },
+              body: {
+                type: "string",
+                description: "The body of the Todo entry."
+              }
+            }
+          }
+        }
+      },
+      delete: {
+        "x-operation-name": "deleteTodo",
+        parameters: [
+          {
+            name: "id",
+            in: "path",
+            description: "The todo ID.",
+            required: true,
+            type: "string",
+            format: "JSON"
+          },
+          {
+            name: "filter",
+            in: "query",
+            description:
+              "The criteria used to narrow down the number of customers returned.",
+            required: false,
+            type: "string",
+            format: "JSON"
+          }
+        ],
+        responses: {
+          204: {}
```

Comment:
That's a matter of interpretation, especially given the many opinions surrounding REST APIs.

If you're a fan of the [Hypermedia As The Engine Of Application State](https://en.wikipedia.org/wiki/HATEOAS) philosophy then what I'm doing would probably not be considered a good thing, since it doesn't lead the user back to some other resource (HATEOAS usually involves returning other URLs so that programs can jump from resource request to resource request using response outputs)

However, returning the code 204 is technically valid, since there's nothing to return (you deleted it!)